server/controller/src/test/java/ai/starwhale/mlops/common/util/TarUtilTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Starwhale, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.starwhale.mlops.common.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import ai.starwhale.mlops.common.TarFileUtil;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.ArchiveOutputStream;
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.io.TempDir;
+import org.springframework.util.ResourceUtils;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class TarUtilTest {
+    static File tar;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        tar = new File(anotherTempDir, "test.tar");
+        tar.createNewFile();
+    }
+
+    @TempDir
+    static File anotherTempDir;
+
+    @Test
+    @Order(1)
+    public void testArchive() throws ArchiveException, IOException {
+        var d = ResourceUtils.getFile("classpath:tar-dir");
+        var files = Files.walk(d.toPath())
+                .filter(path -> path.toFile().isFile())
+                .collect(Collectors.toMap(
+                    (p) -> p.toFile()
+                            .getPath()
+                            .split("tar-dir" + Pattern.quote(System.getProperty("file.separator")))[1], Path::toFile));
```

Comment:
create temporary files instead of resource files. It improves readability. All information is included in the unit test code source.

Reply:
done